pysilcam/tracking/track.py

```diff
@@ -0,0 +1,417 @@
+        meta.attrs['Settings'] = str(settings_dict)
+
+    sctr.process()
+
+
+def make_boxplot(tracksfile):
+    '''
+    given the input tracksfile (-TRACKs.h5), this function will look for all the other -TRACKS.h5 files in the same
+    folder and use this to create summary boxplots of tracked data, which is saved to the General.datafile location
+    specified within the settings contained in the last -TRACKS.h5 file.
+
+    :param tracksfile:
+    :return:
+    '''
+    h5filedir = os.path.split(tracksfile)[0]
+    h5file_list = glob.glob(os.path.join(h5filedir, '*-TRACKS.h5'))
+    dataset_names = [os.path.split(k)[-1].replace('-TRACKS.h5', '') for k in h5file_list]
+
+    tracks = dict()
+    print('* Loading data:')
+    for f, k in zip(h5file_list, dataset_names):
+        print('  ', f)
+        tracks[k] = pd.read_hdf(f, 'Tracking/tracks')
+
+    settings = settings_from_h5(f)
+
+    fig_name = tracksfile.replace('-TRACKS.h5', '-boxplot')
+    plot_boxplot(dataset_names, tracks, settings.PostProcess.pix_size, fig_name)
+    print('* boxplotting finished.')
+
+
+def silctrack():
+    """
+    does tracking
+
+    Usage:
+        silcam-track process <configfile> <datapath> [--offset=<offset>]
+        silcam-track post-process <tracksfile>
+        silcam-track plotting <tracksfile> [--gif=<outputdir>] [<rawdatapath>] [--boxplot]
+    """
+
+    args = docopt(silctrack.__doc__)
+
+    if args['process']:
+        offset = args['--offset']
+        if offset is not None:
+            try:
+                offset = int(offset)
+            except ValueError:
+                print('Expected type int for --offset.')
+                sys.exit(0)
+        else:
+            offset = 0
+
+        track_process(args['<configfile>'], args['<datapath>'],
+                      offset=offset)
+
+    if args['post-process']:
+        print('* Load and process')
+        settings = settings_from_h5(args['<tracksfile>'])
+
+        data, tracks = load_and_process(args['<tracksfile>'],
+                                        settings.PostProcess.pix_size,
+                                        track_length_limit=settings.Tracking.track_length_limit)
+
+        with pd.HDFStore(args['<tracksfile>']) as fh:
+            tracks.to_hdf(fh, 'Tracking/tracks', mode='r+')
+
+        unfiltered_tracks = extract_continuous_tracks(data)
+        with pd.HDFStore(args['<tracksfile>']) as fh:
+            unfiltered_tracks.to_hdf(fh, 'Tracking/unfiltered_tracks', mode='r+')
+
+    if args['plotting']:
+        settings = settings_from_h5(args['<tracksfile>'])
+
+        if args['--gif']:
+
+            if not checkgroup(args['<tracksfile>'], 'Tracking/unfiltered_tracks'):
+                print('* Extracting unfiltered tracks')
+                data = pd.read_hdf(args['<tracksfile>'], 'Tracking/data')
+                unfiltered_tracks = extract_continuous_tracks(data)
```

Comment:
F405: 'extract_continuous_tracks' may be undefined, or defined from star imports: pysilcam.tracking.silcamtracker  <!-- Automatically posted by Lintly -->